citadel-internal-service-connector/src/util.rs

```diff
@@ -48,6 +56,236 @@ impl InternalServiceConnector {
     pub fn split(self) -> (WrappedSink, WrappedStream) {
         (self.sink, self.stream)
     }
+
+    /// Sends a request to register at server running at the given address. Returns a Result with
+    /// an InternalServiceResponse that specifies whether or not the request was successfully sent.
+    pub async fn register<T: Into<SocketAddr>, S: Into<String>, R: Into<SecBuffer>>(
+        &mut self,
+        server_address: T,
+        full_name: S,
+        username: S,
+        proposed_password: R,
+        session_security_settings: SessionSecuritySettings,
+    ) -> Result<InternalServiceResponse, Box<dyn std::error::Error>> {
+        let outbound_request = InternalServiceRequest::Register {
+            request_id: Uuid::new_v4(),
+            server_addr: server_address.into(),
+            full_name: full_name.into(),
+            username: username.into(),
+            proposed_password: proposed_password.into(),
+            session_security_settings,
+            connect_after_register: false,
+        };
+        match self.sink.send(outbound_request).await {
+            Ok(_) => self
```

Comment:
How do we know that this response is the one we want? What if another signal comes through, like a PeerDisconnect or another unrelated signal? We should pattern match against these signals until we get the one we expect, or get the negative signal that implies failure.